### PR TITLE
1596: Add @typescript-eslint rules and fix issues

### PR DESCRIPTION
**Pull request by steffenkleinle**
### Short description

Add @typescript-eslint rules and fix issues (and add a few eslint-disables).

### Proposed changes

- Rewrite .eslintrc to :eslintrc.js
- Use eslint config from integreat-app (and disable a few rules, might be enabled later)
- Enable @tyescript-eslint rules
- Fix most issues

### Side effects

None.

### Testing

That everything still works?

### Resolved issues

Partly fixes: #1596.

### CODE REVIEW

File: administration/src/Router.tsx [hunk 1]
```diff
@@ -80,7 +80,7 @@ const Router = () => {
                 { path: 'region', element: <RegionsController /> },
               ]
             : []),
-          ...(projectConfig.cardStatistics ? [{ path: 'statistics', element: <StatisticsController /> }] : []),
+          ...(projectConfig.cardStatistics.enabled ? [{ path: 'statistics', element: <StatisticsController /> }] : []),
```
Comment (steffenkleinle) [inline administration/src/Router.tsx:83]: Is this correct?

**Comment by seluianova** [comment]
It also fixes this one, right? https://github.com/digitalfabrik/entitlementcard/issues/1296

File: administration/src/Router.tsx (hunk 1 above)
Comment (f1sh1918) [reply administration/src/Router.tsx:86]: yes

File: administration/src/Router.tsx [hunk 2]
```diff
@@ -100,14 +100,7 @@ const Router = () => {
...
     return createBrowserRouter(routes.filter((element): element is RouteObject => element !== null))
-  }, [
-    authData,
-    projectConfig.applicationFeature,
-    signIn,
-    signOut,
-    projectConfig.cardStatistics,
-    projectConfig.cardCreation,
-  ])
+  }, [authData, signIn, signOut, projectConfig])
```
Comment (f1sh1918) [inline administration/src/Router.tsx:103]: is this really working meanwhile with stringify or anything to use objects as deps?

File: administration/src/bp-modules/NavigationBar.tsx [hunk 3]
```diff
@@ -14,11 +14,11 @@ const PrintAwareNavbar = styled(Navbar)`
   }
 `
 
-interface Props {
+type Props = {
```
Comment (f1sh1918) [inline administration/src/bp-modules/NavigationBar.tsx:17]: I think we should define a proper naming within the project. It not has to be fixed here but i think we should name it <component-name>Props or sth similar

**Review by f1sh1918 — CHANGES_REQUESTED** [review]
Cannot start webpack anymore

```
Failed to compile.

Invalid configuration object. Webpack has been initialized using a configuration object that does not match the API schema.
 - configuration.resolve.modules[2] should be a non-empty string.
   -> Folder name or directory path where to find modules.

npm ERR! Lifecycle script `start` failed with error: 
npm ERR! Error: command failed 
npm ERR!   in workspace: administration 
npm ERR!   at location: /Users/afischer/projects/entitlementcard/administration 

Process finished with exit code 1
```

File: administration/src/bp-modules/NavigationBar.tsx (hunk 3 above)
Comment (steffenkleinle) [reply administration/src/bp-modules/NavigationBar.tsx:17]: You mean this should e.g. be `NavigationBarProps`? Agreed, but I don't think there is a rule for this, we just have to take care of this manually/during reviewing. Do you want me to create an issue for this? Or should we do this on the fly?

File: administration/src/Router.tsx (hunk 2 above)
Comment (steffenkleinle) [reply administration/src/Router.tsx:103]: This should work (and is as far as I can see). But it depends. If the object is recreated every render, e.g. as in [this example](https://react.dev/reference/react/useEffect#removing-unnecessary-object-dependencies) of the react docs, the hook is executed again on every render.
However, if the object is not redefined/does not change on every render, this works just fine. This is e.g. the case if it is a value from `useState`, `useMemo`, `useContext` or defined outside of a react component, which is the case here.

**Comment by steffenkleinle** [comment]
> Cannot start webpack anymore
> 
> ```
> Failed to compile.
> 
> Invalid configuration object. Webpack has been initialized using a configuration object that does not match the API schema.
>  - configuration.resolve.modules[2] should be a non-empty string.
>    -> Folder name or directory path where to find modules.
> 
> npm ERR! Lifecycle script `start` failed with error: 
> npm ERR! Error: command failed 
> npm ERR!   in workspace: administration 
> npm ERR!   at location: /Users/afischer/projects/entitlementcard/administration 
> 
> Process finished with exit code 1
> ```

Ohhh... funny, this was caused by this little change:
``` ts
      modules: ['node_modules', paths.appNodeModules].concat(modules.additionalModulePaths || []),
```
to
``` ts
      modules: ['node_modules', paths.appNodeModules].concat(modules.additionalModulePaths ?? []),
```

Should be fixed now @f1sh1918, can you please test this again? Thanks.

File: .idea/dataSources.xml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
   <component name="DataSourceManagerImpl" format="xml" multifile-model="true">
-    <data-source source="LOCAL" name="ehrenamtskarte@localhost" uuid="d83188d7-bad5-4b46-b4d7-13cabcdaf7c7">
+    <data-source source="LOCAL" name="ehrenamtskarte@db-postgis" uuid="d83188d7-bad5-4b46-b4d7-13cabcdaf7c7">
```
[the remark this answers is not in the document]
Comment (steffenkleinle) [reply .idea/dataSources.xml]: Because it keeps changing for me. Reverted it neverthelesss

**Review by ztefanie — COMMENTED** [review]
Looks good :+1:
Thanks for the adjustments. Not tested again.

**Review by ztefanie — APPROVED** [review]

**Review by f1sh1918 — APPROVED** [review]
Looks good 👍

**Pull request merged**